### PR TITLE
WFLY-18002 ExpirationMetaData.isExpired() should return true if expiration time is exactly now.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18002

### CODE REVIEW

File: clustering/ee/spi/src/main/java/org/wildfly/clustering/ee/expiration/ExpirationMetaData.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public interface ExpirationMetaData extends Expiration {
     default boolean isExpired() {
         if (this.isImmortal()) return false;
         Instant lastAccessedTime = this.getLastAccessTime();
-        return (lastAccessedTime != null) ? lastAccessedTime.plus(this.getTimeout()).isBefore(Instant.now()) : false;
+        return (lastAccessedTime != null) ? !lastAccessedTime.plus(this.getTimeout()).isAfter(Instant.now()) : false;
```
Comment (kabir) [inline clustering/ee/spi/src/main/java/org/wildfly/clustering/ee/expiration/ExpirationMetaData.java:40]: I might be misunderstanding something but won't this still return false if lastAccessedTime+timeout == now?

Comment (pferraro) [reply clustering/ee/spi/src/main/java/org/wildfly/clustering/ee/expiration/ExpirationMetaData.java:40]: I don't think so.
Current behavior is:
```java
instant.isBefore(now)
```
i.e. instant < now.  However, we want this to be inclusive, such that instant <= now

```java
instant.isAfter(now)
```
means instant > now, thus the inverse:
```java
!instant.isAfter(now)
```
means instant <= now, which is what we want.

Comment (bstansberry) [reply clustering/ee/spi/src/main/java/org/wildfly/clustering/ee/expiration/ExpirationMetaData.java:40]: 👍 It's the subtle added '!' in the diff that's key.

**Review by kabir — APPROVED** [review]

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @pferraro and @kabir

**Pull request merged**